extensions/pkg/controller/worker/genericactuator/interface.go

```diff
@@ -61,18 +61,6 @@ type WorkerDelegate interface {
 	PostDeleteHook(context.Context) error
 }
 
-// WorkerCredentialsDelegate is an interface that can optionally be implemented to be
-// used during the Worker reconciliation to keep all machine class secrets up to date.
-// DEPRECATED: extensions should instead provide credentials only (!) via the machine class field .spec.credentialsSecretRef
-// referencing the Worker's secret reference (spec.SecretRef). This way all machine classes
-// reference the same secret - there is no need anymore to update all machine class secrets.
-// please see [here](https://github.com/gardener/machine-controller-manager/pull/578) for more details
-type WorkerCredentialsDelegate interface {
-	// GetMachineControllerManagerCloudCredentials should return the IaaS credentials
-	// with the secret keys used by the machine-controller-manager.
-	GetMachineControllerManagerCloudCredentials(context.Context) (map[string][]byte, error)
```

Comment:
FYI @MartinWeindel @briantopping provider-vsphere is still using this `GetMachineControllerManagerCloudCredentials` func - https://github.com/gardener/gardener-extension-provider-vsphere/blob/19baf64322c391bad9ae73f5928c29f45220d5f4/pkg/controller/worker/machine_controller_manager.go#L83-L87